internal/jennies/java/rawtypes.go

```diff
@@ -0,0 +1,327 @@
+package java
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/jennies/common"
+	"github.com/grafana/cog/internal/tools"
+)
+
+type RawTypes struct {
+	config  Config
+	imports *common.DirectImportMap
+
+	typeFormatter *typeFormatter
+}
+
+func (jenny RawTypes) JennyName() string {
+	return "JavaBuilder"
+}
+
+func (jenny RawTypes) Generate(context common.Context) (codejen.Files, error) {
+	files := make(codejen.Files, 0)
+	jenny.imports = NewImportMap()
+	jenny.typeFormatter = createFormatter(context)
+
+	for _, schema := range context.Schemas {
+		output, err := jenny.genFilesForSchema(schema)
+		if err != nil {
+			return nil, err
+		}
+
+		files = append(files, output...)
+	}
+
+	return files, nil
+}
+
+func (jenny RawTypes) genFilesForSchema(schema *ast.Schema) (codejen.Files, error) {
+	files := make(codejen.Files, 0)
+	scalars := make(map[string]ast.ScalarType)
+
+	packageMapper := func(pkg string, class string) string {
+		if pkg == schema.Package {
+			return ""
+		}
+
+		return jenny.imports.Add(class, pkg)
+	}
+
+	jenny.typeFormatter = jenny.typeFormatter.withPackageMapper(packageMapper)
+
+	for _, object := range schema.Objects {
+		jenny.imports = NewImportMap()
+		if object.Type.IsMap() {
+			continue
+		}
+		if object.Type.IsScalar() {
```

Comment:
It looks like you're treating these objects are constants: that's not necessarily correct.

This check only tells us that this object aliases a scalar type. For actual constants, we need to do something like `object.Type.IsScalar() && object.Type.AsScalar().IsConcrete()`